base/compiler/abstractinterpretation.jl

```diff
@@ -1057,6 +1058,31 @@ function abstract_call_known(interp::AbstractInterpreter, @nospecialize(f),
     return abstract_call_gf_by_type(interp, f, argtypes, atype, sv, max_methods)
 end
 
+function abstract_call_opaque_closure(interp::AbstractInterpreter, closure::PartialOpaque, argtypes::Vector{Any}, sv::InferenceState)
+    return CallMeta(Any, nothing)
+end
+
+function most_general_argtypes(closure::PartialOpaque)
```

Comment:
This seems like incomplete copy of `matching_cache_argtypes`. Can we use that instead (refactoring as needed)?